GitCommands/Git/ExecutableExtensions.cs

```diff
@@ -363,9 +374,13 @@ string ComposeOutput()
 
             var output = outputEncoding.GetString(outputBuffer.GetBuffer(), 0, (int)outputBuffer.Length);
             var error = outputEncoding.GetString(errorBuffer.GetBuffer(), 0, (int)errorBuffer.Length);
-
             var exitCode = await process.WaitForExitAsync();
 
+            if (cache is not null && await exitTask == 0)
+            {
+                cache.Add(arguments, outputBuffer.ToArray(), errorBuffer.ToArray());
```

Comment:
We must be careful not to cache too much.
I guess that I have seen diffs with Index and WD not updated correctly until I reloaded everything in the past month.

Reply:
Sidenote to this PR:
If something is cached, a refresh will not update automatically, the cache contents is kept.
So what you saw is likely that the file lists or the contents of a diff is not updated automatically at changes.
I would like to have notification about changes in the diff tab and FormCommit similar to the Browse refresh (FileStatusMonitor) to detect this.

However, I want a cache for git-status too, but that is a new functionality, the cache will need quite heavy invalidation policy...